## Task 1: Permutations and Combinations

### Problem Description

We have **12 cups of tea**:
- **6 cups** have milk added first.
- **6 cups** have tea added first.

We want to calculate the following probabilities:
1. **Probability of guessing all 6 milk-first cups correctly** (i.e. 0 errors).
2. **Probability of guessing at least 5 milk-first cups correctly** (i.e. ≤ 1 error).
3. **Probability of guessing at least 4 milk-first cups correctly** (i.e. ≤ 2 errors), and a discussion of why allowing 2 errors is too lenient.

In [8]:
# Part 1

from math import comb

# Part 1: Probability of Guessing All 6 Correctly

# 1. Total ways to pick 6 cups out of 12.
total_combinations = comb(12, 6)

# 2. Exactly 1 way matches the true set of 6 milk-first cups.
correct_combination = 1

# 3. Probability (as a percentage) of selecting the correct 6.
probability_all_correct = (correct_combination / total_combinations) * 100

print("Total number of ways to choose 6 from 12:", total_combinations)
print(f"Probability of guessing all 6 correctly: {probability_all_correct:.4f}%")

Total number of ways to choose 6 from 12: 924
Probability of guessing all 6 correctly: 0.1082%


In [9]:
# Part 2: Probability of Making At Most One Error (at least 5 correct cups)

# 4. Ways to choose exactly 5 correct out of 6 milk-first cups,
#    and 1 incorrect from the 6 tea-first cups.
correct_5_wrong_1 = comb(6, 5) * comb(6, 1)

# 5. Total ways for at most 1 error = exactly 6 correct + exactly 5 correct.
correct_6_or_5 = correct_combination + correct_5_wrong_1

# 6. Probability of at most 1 error
probability_one_error = (correct_6_or_5 / total_combinations) * 100

print(f"Number of ways to select exactly 5 correct and 1 wrong: {correct_5_wrong_1}")
print(f"Probability of at least 5 correct (≤ 1 error): {probability_one_error:.4f}%")

Number of ways to select exactly 5 correct and 1 wrong: 36
Probability of at least 5 correct (≤ 1 error): 4.0043%


In [10]:
# Part 3: Probability of Making Exactly 2 Errors (at least 4 correct cups)

# 7. Ways to select exactly 4 correct cups (out of 6) and 2 incorrect (out of 6).
correct_4_wrong_2 = comb(6, 4) * comb(6, 2)

# 8. Probability of making exactly 2 errors
probability_two_errors = (correct_4_wrong_2 / total_combinations) * 100

print(f"Number of ways to select exactly 4 correct and 2 wrong: {correct_4_wrong_2}")
print(f"Probability of at least 4 correct (2 errors): {probability_two_errors:.4f}%")

Number of ways to select exactly 4 correct and 2 wrong: 225
Probability of at least 4 correct (2 errors): 24.3506%


### Discussion (Would We Accept Two Errors)?

- **Probability of Guessing All 6 Correctly (0 errors)**  
  This probability is quite low (around 0.11%). In statistical terms, if a person achieves this result purely by guessing, it would be considered highly unlikely. Consequently, this outcome could be viewed as evidence of actual skill if repeated consistently.


- **Probability of Guessing At Most 1 Error (≤ 1 error)**  
  Allowing exactly one mistake increases the chance to around 4%. Although higher, it is still below the conventional **5% significance level**. Hence, whilst not as stringent as requiring a perfect score, it remains relatively difficult to achieve through random guessing alone.


- **Allowing 2 Errors (≤ 2 errors)**  
  By permitting two mistakes, the success probability rises sharply to around 24%. This means nearly one in four random guessers could meet this threshold. Since it far exceeds 5%, it would not be considered **statistically significant**, and thus provides little evidence of an actual ability to discern which cups had milk added first.
  
 
#### Conclusion:

In summary, lower probabilities indicate stricter tests and a greater likelihood that a successful outcome is due to skill rather than guesswork. Once the probability of success climbs above 5%, the result often stops being considered “significant” in a traditional statistical sense. If we were to accept two errors it would make it much more likely that a person could pass the test by guessing alone, rather than demonstrating any real skill or ability. Therefore we would not accept two errors

### References

1. Fisher, R. A. (1935). *The Design of Experiments.* Oliver & Boyd.